### PR TITLE
Add OceanPop

### PR DESCRIPTION
**Pull request by sharkwouter**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link: https://github.com/sharkwouter/oceanpop**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

I'd like to add an open source puzzle game called OceanPop to the Flathub. I developed this game over the course of about half a year and I think I managed to make quite a polished release out of it. It should build for all platforms the Flathub supports.

I had to use the `--device=all` permission to get gamepads to work unfortunately.

### CODE REVIEW

File: io.github.sharkwouter.OceanPop.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+app-id: io.github.sharkwouter.OceanPop
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
```
Comment (barthalion) [inline io.github.sharkwouter.OceanPop.yaml:3]:
```suggestion
runtime-version: '22.08'
```

**Comment by barthalion** [comment]
bot, build io.github.sharkwouter.OceanPop

File: io.github.sharkwouter.OceanPop.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  - name: jsoncpp
+    buildsystem: meson
+    config-opts:
+      - "--default-library=shared"
+    sources:
+      - type: archive
+        url: https://github.com/open-source-parsers/jsoncpp/archive/refs/tags/1.9.5.tar.gz
+        sha256: f409856e5920c18d0c2fb85276e24ee607d2a09b5e7d5f0a371368903c275da2
+  - name: oceanpop
+    buildsystem: cmake
```
Comment (hfiguiere) [inline io.github.sharkwouter.OceanPop.yaml:22]:
```suggestion
    buildsystem: cmake-ninja
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]
Build fail because
```
error: /srv/buildbot/worker/build-x86_64-6/build/builddir/export/share/icons/hicolor/128x128/apps/io.github.sharkwouter.OceanPop.png is not a valid icon: Expected a square icon but got: 128x127
```

The icon is supposed to be square, and appstream enforce this.

**Comment by sharkwouter** [comment]
bot, build io.github.sharkwouter.OceanPop

**Comment by sharkwouter** [comment]
I had no idea the icon was 1 pixel off, that was not intended. I fixed it and created a new tag with the fix.

**Comment by sharkwouter** [comment]
bot, build io.github.sharkwouter.OceanPop

**Comment by sharkwouter** [comment]
bot, build io.github.sharkwouter.OceanPop

**Comment by sharkwouter** [comment]
Closing in favor of #3636

**Comment by hfiguiere** [comment]
WHY? it makes review harder

**Comment by sharkwouter** [comment]
I wanted to change the id from io.github.sharkwouter.OceanPop to io.github.sharkwouter.oceanpop, which requires a new branch. Sorry about that. It should finally work now.

**Comment by hfiguiere** [comment]
not it doesn't require a new branch. branch name is irrelevant.

**Comment by sharkwouter** [comment]
Oh, I see

**Comment by sharkwouter** [comment]
Turns out I cannot actually reopen this issue, since I did a force push. I'm sorry :(

**Comment by barthalion** [comment]
Yeah, you need to open a new PR. This is unfortunately how GitHub works.

**Pull request closed without merging**